src/test/java/org/apache/ibatis/submitted/custom_collection_handling/ImmutableCustomCollection.java

```diff
@@ -0,0 +1,62 @@
+/**
+ *    Copyright 2009-2018 the original author or authors.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package org.apache.ibatis.submitted.custom_collection_handling;
+
+import java.util.Collection;
+import java.util.List;
+
+public class ImmutableCustomCollection<T> extends CustomCollection<T> {
```

Comment:
Hi @h3adache ,
Could you make this class not to extend `CustomCollection` ?
An immutable collection class usually does not extend mutable collection (I think) and I would like to see how it fits in this.